luda-adventure/src/app/game/update/handle_character_movement_on_key_event.rs

```diff
@@ -0,0 +1,47 @@
+use crate::app::game::*;
+use namui::prelude::*;
+
+impl Game {
+    pub fn handle_character_movement_on_key_event(&mut self, current_time: Time) {
+        if let Some((_character, (_player_character, positioner))) = self
+            .ecs_app
+            .query_entities_mut::<(&PlayerCharacter, &mut Positioner)>()
+            .first_mut()
+        {
+            let character_velocity = get_character_velocity_from_key_state();
+            let character_velocity_has_not_changed = character_velocity.x * 1.ms()
+                == self.state.character.last_velocity.x * 1.ms()
+                && character_velocity.y * 1.ms() == self.state.character.last_velocity.y * 1.ms();
```

Comment:
Why do you multiply 1.ms() instead of just comparing Per struct?

Reply:
Just because `Ord` is not implemented for `Per`.

Reply:
Then what about impl that?

Reply:
Good. I will try that

Reply:
[44d77a9](https://github.com/NamseEnt/namseent/pull/359/commits/44d77a943a2abfe3f318d070656c4768d20aa224)

Reply:
#379